# NCMHap
By Fatemeh Zamani, Email: f_zamani@znu.ac.ir.

NCMHap is free; you can redistribute it and/or modify it, to reconstruct haplotype sequences for diploid, polyploid.

NCMHap will consider the Neutrosophic C-Means clustering algorithm, as a proposed solution to the haplotype reconstruction problem.

# Requirement:

NCMHap is implemented in MATLAB R2014a and above versions.

The main program for diploid case is NCMHap.m, and for polyploid case is NCMHap-poly.m.

For diploid case, We need Dr. F. Geraci simulated dataset (filippo.geraci@iit.cnr.it) that refer to his paper: Geraci, F., 2010. A comparison of several algorithms for the single individual SNP haplotyping reconstruction problem. Bioinformatics 26, 2217-2225.

For polyploid case, The simulated data set corresponds to what is in this papers:


1. Berger, E., et al., Haptree: A novel bayesian framework for single individual polyplotyping using ngs data. PLoS computational biology, 2014. 10(3): p. e1003502.


2. Mazrouee, S. and W. Wang. HapColor: A graph coloring framework for polyploidy phasing. in 2015 IEEE International Conference on Bioinformatics and Biomedicine (BIBM). 2015. IEEE.


3. Hashemi, A., B. Zhu, and H. Vikalo, Sparse tensor decomposition for haplotype assembly of diploids and polyploids. BMC genomics, 2018. 19(4): p. 191.


4. Motazedi, E., et al., TriPoly: haplotype estimation for polyploids using sequencing data of related individuals. Bioinformatics, 2018. 34(22): p. 3864-3872.


The polyploid dataset will be shared if needed.
